src/librustc_typeck/check/regionck.rs

```diff
@@ -1795,7 +1795,7 @@ impl<'a, 'gcx, 'tcx> RegionCtxt<'a, 'gcx, 'tcx> {
                     let (outlives, _) =
                         self.replace_late_bound_regions_with_fresh_var(
                             span,
-                            infer::AssocTypeProjection(projection_ty.item_name(self.tcx)),
+                            infer::AssocTypeProjection(self.tcx.associated_item(projection_ty.item_def_id).name),
```

Comment:
@eddyb putting a DefId in AssocTypeProjection means we lose the name here: https://github.com/rust-lang/rust/blob/c7104be/src/librustc/infer/error_reporting/mod.rs#L902

How, in that context, do I get the name from the DefId?